lib/client/api.go

```diff
@@ -3299,6 +3302,11 @@ func (tc *TeleportClient) Login(ctx context.Context) (*Key, error) {
 	// Perform the ALPN test once at login.
 	tc.TLSRoutingConnUpgradeRequired = client.IsALPNConnUpgradeRequired(tc.WebProxyAddr, tc.InsecureSkipVerify)
 
+	authMaxSessTTL := time.Duration(pr.Auth.MaxSessionTTL)
+	if !tc.CliTTL {
+		tc.KeyTTL = authMaxSessTTL
+	}
+
```

Comment:
This and the removal of the default setting for `cf.MinsToLive` avoids an extra flag for "user has specified `--ttl`".
```suggestion
	if c.KeyTTL == 0 {
		c.KeyTTL = time.Duration(pr.Auth.MaxSessionTTL)
	}
	// DELETE IN some version where the auth is guaranteed to send us a valid MaxSessionTTL or the auth is guaranteed to interpret 0 duration as the auth's default?
	if c.KeyTTL == 0 {
		c.KeyTTL = apidefaults.CertDuration
	}
```